scripts/bert/create_pretraining_data.py

```diff
@@ -416,15 +417,51 @@ def create_instances_from_document(x):
 
 MaskedLmInstance = collections.namedtuple('MaskedLmInstance',
                                           ['index', 'label'])
+def find_chinese_whole_word(universal_tokenizer, tokens, chs_tokenizer=None):
+    """find chinese characters in tokenized results, and tag them in BIO format with chinese "
+       tokenizer.
+    Parameters
+    ----------
+    tokenizer : BERTTokenizer
+    tokens : list of str
+        bert tokenized tokens: e.g.: ['你', '好', '12', '##3']
 
+    Returns
+    -------
+    A tuple of list: tokens, chs_token_bio;
+        e.g.:   tokens = ['你', '好', '12', '##3']
+                chs_token_bio = ['B', 'I', 'O', 'O']
+    """
+    chs_token_bio = ['O']*len(tokens)
+    if isinstance(universal_tokenizer, BERTTokenizer) and chs_tokenizer is not None:
+        is_chinese_list = ['1' if len(token) == 1 and
+                           universal_tokenizer.basic_tokenizer._is_chinese_char(ord(token))
+                           else '0'
+                           for token in tokens]
+        patten = ''.join(is_chinese_list)
+        for re_find in re.finditer('[1]{1,}', patten):
+            chs_start, chs_end = (re_find.start(), re_find.end())
+            cut_results = chs_tokenizer(''.join(tokens[chs_start: chs_end]))
+            idx = 0
+            for cut_token in cut_results:
+                for i in range(len(cut_token)):
+                    if i == 0:
+                        chs_token_bio[chs_start + idx] = 'B'
+                    else:
+                        chs_token_bio[chs_start + idx] = 'I'
+                    idx += 1
+        assert len(tokens) == len(chs_token_bio)
+    return tokens, chs_token_bio
 
 def create_masked_lm_predictions(tokens, masked_lm_prob, max_predictions_per_seq,
                                  whole_word_mask, vocab, tokenizer,
                                  _MASK_TOKEN, _CLS_TOKEN, _SEP_TOKEN):
     """Creates the predictions for the masked LM objective."""
     cand_indexes = []
```

Comment:
Can we make whole word masking for Chinese optional, by adding a separate flag  (cn_whole_word_mask) to create_pretraining_data.py, `create_training_instances`, run_pretraining_hvd.py? 

```
if cn_whole_word:
    chs_tokenizer = JiebaTokenizer()
    tokens, chs_token_bio = find_chinese_whole_word(tokenizer, tokens, chs_tokenizer)
```

Reply:
OK. Added.